tests/tom-swifty-test/XmlReflectionTests/DynamicXmlTests.cs

```diff
@@ -1400,5 +1400,102 @@ func whoami(a: (Self)->()) -> ()
 			Assert.IsNotNull (proto, "no protocol");
 			Assert.IsTrue (proto.HasDynamicSelf, "no dynamic self");
 		}
+
+		[Test]
+		public void SelfIsInAssociatedTypesArg ()
+		{
+			var code = @"
+public protocol Simple {
+	func whoami (a: Self)
+}
+";
+			var module = ReflectToModules (code, "SomeModule").Find (m => m.Name == "SomeModule");
+			Assert.IsNotNull (module, "module is null");
+			var proto = module.Protocols.Where (p => p.Name == "Simple").FirstOrDefault ();
+			Assert.IsNotNull (proto, "no protocol");
+			Assert.AreEqual (1, proto.AssociatedTypes.Count, "assoc type count mismatch");
+			Assert.AreEqual ("Self", proto.AssociatedTypes [0].Name);
+		}
+
+		[Test]
+		public void SelfIsInAssociatedTypesReturn ()
+		{
+			var code = @"
+public protocol Simple {
+	func whoami () -> Self
+}
+";
+			var module = ReflectToModules (code, "SomeModule").Find (m => m.Name == "SomeModule");
+			Assert.IsNotNull (module, "module is null");
+			var proto = module.Protocols.Where (p => p.Name == "Simple").FirstOrDefault ();
```

Comment:
No need to change this but FYI FirstOrDefault can take a where predicate. I'm pretty sure you can write this as:

` var proto = module.Protocols.FirstOrDefault (p => p.Name == "Simple");`